File: frontend/src/types/agent.ts
/**
 * Agent-related TypeScript types matching backend schemas.
 */

/**
 * Agent status enum values.
 */
export type AgentStatus = 'draft' | 'active' | 'deprecated';

/**
 * Author information nested in agent response.
 */
export interface AuthorInfo {
  id: string;
  name: string;
  email: string;
}

/**
 * Organization information nested in agent response.
 */
export interface OrganizationInfo {
  id: string;
  name: string;
}

/**
 * Manager information nested in agent response.
 */
export interface ManagerInfo {
  id: string;
  name: string;
  email: string;
}

/**
 * Agent data returned from API.
 */
export interface Agent {
  id: string;
  name: string;
  description: string | null;
  author_id: string;
  author: AuthorInfo | null;
  current_version_id: string | null;
  status: AgentStatus;
  tags: string[];
  department: string | null;
  usage_notes: string | null;
  organization_id: string | null;
  organization: OrganizationInfo | null;
  manager_id: string | null;
  manager: ManagerInfo | null;
  version_count: number;
  is_running: boolean;
  created_at: string;
  updated_at: string;
}

/**
 * Request body for creating a new agent.
 */
export interface AgentCreate {
  name: string;
  description?: string | null;
  tags?: string[];
  department?: string | null;
  usage_notes?: string | null;
}

/**
 * Request body for updating an agent.
 */
export interface AgentUpdate {
  name?: string;
  description?: string | null;
  status?: AgentStatus;
  tags?: string[];
  department?: string | null;
  usage_notes?: string | null;
  organization_id?: string | null;
  manager_id?: string | null;
}

/**
 * Paginated list response for agents.
 */
export interface AgentListResponse {
  data: Agent[];
  total: number;
}

/**
 * Query parameters for listing agents.
 */
export interface AgentListParams {
  status?: AgentStatus;
  search?: string;
  skip?: number;
  limit?: number;
}

File: backend/app/schemas/agent.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime
from typing import Optional
from app.models.agent import AgentStatus


class AgentCreate(BaseModel):
    name: str
    description: Optional[str] = None
    tags: list[str] = []
    department: Optional[str] = None
    usage_notes: Optional[str] = None


class AgentUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    status: Optional[AgentStatus] = None
    tags: Optional[list[str]] = None
    department: Optional[str] = None
    usage_notes: Optional[str] = None
    organization_id: Optional[UUID] = None
    manager_id: Optional[UUID] = None


class AuthorInfo(BaseModel):
    """Nested author information."""
    id: UUID
    name: str
    email: str

    class Config:
        from_attributes = True


class OrganizationInfo(BaseModel):
    """Nested organization information."""
    id: UUID
    name: str

    class Config:
        from_attributes = True


class ManagerInfo(BaseModel):
    """Nested manager information."""
    id: UUID
    name: str
    email: str

    class Config:
        from_attributes = True


class AgentResponse(BaseModel):
    id: UUID
    name: str
    description: Optional[str]
    author_id: UUID
    author: Optional[AuthorInfo] = None
    current_version_id: Optional[UUID]
    status: AgentStatus
    tags: list[str]
    department: Optional[str]
    usage_notes: Optional[str]
    organization_id: Optional[UUID] = None
    organization: Optional[OrganizationInfo] = None
    manager_id: Optional[UUID] = None
    manager: Optional[ManagerInfo] = None
    version_count: int = 0
    is_running: bool = False
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class AgentListResponse(BaseModel):
    data: list[AgentResponse]
    total: int
